Signal/src/view controllers/CallViewController.swift

```diff
@@ -132,19 +139,36 @@ class CallViewController: UIViewController, CallObserver {
 
         // Subscribe for future call updates
         call.addObserverAndSyncState(observer: self)
+
+        Environment.getCurrent().callService.addObserverAndSyncState(observer:self)
     }
 
+    // MARK: - Create Views
+
     func createViews() {
         // Dark blurred background.
         let blurEffect = UIBlurEffect(style: .dark)
         blurView = UIVisualEffectView(effect: blurEffect)
         self.view.addSubview(blurView)
 
+        // Create the video views first, as they are under the other views.
+        createVideoViews()
+
         createContactViews()
         createOngoingCallControls()
         createIncomingCallControls()
     }
 
+    func createVideoViews() {
+        remoteVideoView = RTCEAGLVideoView()
+        remoteVideoView.delegate = self
+        localVideoView = RTCCameraPreviewView()
```

Comment:
Really? I would be less surprised if it were when peerConnection is initialized, but not until ringing starts seems arbitrary. What are we doing that fires the request?